templates/nuxt/store/crud.js

```diff
@@ -0,0 +1,273 @@
+import Vue from 'vue';
+import { getField, updateField } from 'vuex-map-fields';
+import remove from 'lodash/remove';
+import SubmissionError from '../error/SubmissionError';
+
+const initialState = () => ({
+  allIds: [],
+  byId: {},
+  created: null,
+  deleted: null,
+  error: "",
+  isLoading: false,
+  resetList: false,
+  selectItems: null,
+  totalItems: 0,
+  updated: null,
+  view: null,
+  violations: null
+});
+
+const handleError = (commit, e) => {
+  commit(ACTIONS.TOGGLE_LOADING);
+
+  if (e instanceof SubmissionError) {
+    commit(ACTIONS.SET_VIOLATIONS, e.errors);
+    // eslint-disable-next-line
+    commit(ACTIONS.SET_ERROR, e.errors._error);
+
+    return Promise.reject(e);
+  }
+
+  // eslint-disable-next-line
+  commit(ACTIONS.SET_ERROR, e.message);
+
+  return Promise.reject(e);
+};
+
+export const ACTIONS = {
+  ADD: 'ADD',
+  RESET_CREATE: 'RESET_CREATE',
+  RESET_DELETE: 'RESET_DELETE',
+  RESET_LIST: 'RESET_LIST',
+  RESET_SHOW: 'RESET_SHOW',
+  RESET_UPDATE: 'RESET_UPDATE',
+  SET_CREATED: 'SET_CREATED',
+  SET_DELETED: 'SET_DELETED',
+  SET_ERROR: 'SET_ERROR',
+  SET_SELECT_ITEMS: 'SET_SELECT_ITEMS',
+  SET_TOTAL_ITEMS: 'SET_TOTAL_ITEMS',
+  SET_UPDATED: 'SET_UPDATED',
+  SET_VIEW: 'SET_VIEW',
+  SET_VIOLATIONS: 'SET_VIOLATIONS',
+  TOGGLE_LOADING: 'TOGGLE_LOADING'
+};
+
+export default function makeCrudModule({
+  normalizeRelations = x => x,
+  resolveRelations = x => x,
```

Comment:
What is the difference between those?
Am I correct, that I should use them to expand the related items from their iri to the full items?
Is there an example how to use them?

Reply:
You are right, one is for normalize some fields before sending them to the api and the other one is for retrieving some data when having only the iri for an item.

I don't fully implement this cause it could be hard to understand and it relies heavily on the Vuex (you need a store for each relation).

This implementation was inspired by this [post](https://markus.oberlehner.net/blog/generic-content-vuex-modules/), you'll find some examples

Reply:
Ok, thanks for the explanation!